src/shogun/clustering/Hierarchical.cpp

```diff
@@ -212,8 +216,94 @@ SGMatrix<int32_t> CHierarchical::get_cluster_pairs()
 	return SGMatrix<int32_t>(pairs,2,merges, false);
 }
 
-
 void CHierarchical::store_model_features()
 {
-	/* TODO. Currently does nothing since apply methods are not implemented. */
+	CDenseFeatures<float64_t>* rhs =
+	    dynamic_cast<CDenseFeatures<float64_t>*>(distance->get_rhs());
+
+	int32_t num_vectors = rhs->get_num_vectors();
+	int32_t num_features = rhs->get_num_features();
+
+	int32_t* n_cluster_samples = SG_MALLOC(int32_t, num_vectors + merges);
+	SGVector<int32_t>::fill_vector(n_cluster_samples, num_vectors + merges, 0);
+
+	SGMatrix<float64_t>* null_matrix =
+	    new SGMatrix<float64_t>(num_features, num_vectors + merges);
```

Comment:
Why do you allocate memory on the heap here? Shouldn't it be `SGMatrix<float64_t> null_matrix(num_features, num_vectors + merges);`?

Reply:
I felt that if the `num_features` and `num_vectors` are too large, then it might be preferable to allocate the memory on heap. That's why I did that.

Reply:
Ohh. I made an oopsie. But, is it necessary to change it?

Reply:
Made the change anyways. Thanks for pointing it out :)